agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -394,6 +394,42 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
     return true;
 }
 
+bool slave_thread::start_monitoring(const beerocks_message::sClientMonitoringParams &params,
+                                    beerocks::beerocks_header &beerocks_header)
+{
+    std::string client_mac              = network_utils::mac_to_string(params.mac);
+    std::string client_bridge_4addr_mac = network_utils::mac_to_string(params.bridge_4addr_mac);
+    std::string client_ip               = network_utils::ipv4_to_string(params.ipv4);
+
+    LOG(DEBUG) << "START_MONITORING_REQUEST: mac=" << client_mac << " ip=" << client_ip
+               << " bridge_4addr_mac=" << client_bridge_4addr_mac
+               << " vap_id=" << std::to_string(params.vap_id)
+               << " is_ire=" << std::to_string(params.is_ire);
+
+    if (params.is_ire) {
+        auto request_out = message_com::create_vs_message<
+            beerocks_message::cACTION_APMANAGER_CLIENT_IRE_CONNECTED_NOTIFICATION>(cmdu_tx);
+        if (!request_out) {
+            LOG(ERROR) << "Failed building ACTION_APMANAGER_CLIENT_IRE_CONNECTED_NOTIFICATION "
+                          "message!";
+            return false;
+        }
+        request_out->mac() = params.mac;
+        message_com::send_cmdu(ap_manager_socket, cmdu_tx);
+    }
+
+    //notify monitor
+    auto request_out = message_com::create_vs_message<
+        beerocks_message::cACTION_MONITOR_CLIENT_START_MONITORING_REQUEST>(cmdu_tx,
+                                                                           beerocks_header.id());
```

Comment:
This is the only use of beerocks_header, so I don't think it makes much sense to pass the entire header.

For the new call, the id parameter can just be zero.